### ISSUE TITLE
Restarting Harbor via 'docker-compose up' deletes all repositories

### ISSUE THREAD

**Issue by klixxezitz** [vmware/harbor#3376]
First of all the needed information
1) Versions of Docker and Docker-Compose
- Docker version 1.12.5, build 7392c3b
- docker-compose version 1.8.0, build 94f7016

2)Config files of harbor, you can get them by packaging "harbor.cfg" and files in the same directory, including subdirectory.
- Will be included later if really needed

3)Log files, you can get them by package the /var/log/harbor/ .
See below

Hej all,

We are still running harbor 0.5.0 here and faced a very ugly problem today when doing a 'docker-compose down' and 'docker-compose up' afterwards.

When harbor started up, we got the following output to stdout:

Attaching to harbor-log, harbor-db, registry, harbor-ui, nginx, harbor-jobservice
harbor-db     | 2017-10-13 13:52:04 0 [Note] mysqld (mysqld 5.6.34) starting as process 1 ...
harbor-db     | 2017-10-13 13:52:04 1 [Note] Plugin 'FEDERATED' is disabled.
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Using atomics to ref count buffer pool pages
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: The InnoDB memory heap is disabled
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Memory barrier is not used
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Compressed tables use zlib 1.2.8
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Using Linux native AIO
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Using CPU crc32 instructions
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Initializing buffer pool, size = 128.0M
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Completed initialization of buffer pool
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Highest supported file format is Barracuda.
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: 128 rollback segment(s) are active.
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: Waiting for purge to start
harbor-db     | 2017-10-13 13:52:04 1 [Note] InnoDB: 5.6.34 started; log sequence number 4058836
harbor-db     | 2017-10-13 13:52:04 1 [Note] Server hostname (bind-address): '*'; port: 3306
harbor-db     | 2017-10-13 13:52:04 1 [Note] IPv6 is available.
harbor-db     | 2017-10-13 13:52:04 1 [Note]   - '::' resolves to '::';
harbor-db     | 2017-10-13 13:52:04 1 [Note] Server socket created on IP: '::'.
harbor-db     | 2017-10-13 13:52:04 1 [Warning] 'proxies_priv' entry '@ root@c1e2ab2af9bd' ignored in --skip-name-resolve mode.
harbor-db     | 2017-10-13 13:52:04 1 [Note] Event Scheduler: Loaded 0 events
harbor-db     | 2017-10-13 13:52:04 1 [Note] mysqld: ready for connections.
harbor-db     | Version: '5.6.34'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)
registry      | time="2017-10-13T13:52:04.402079301Z" level=info msg="configuring endpoint harbor (http://ui/service/notifications), timeout=3s, headers=map[]" go.version=go1.6.3 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:04.405960686Z" level=info msg="redis not configured" go.version=go1.6.3 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:04.417228544Z" level=info msg="debug server listening localhost:5001"
registry      | time="2017-10-13T13:52:04.449487422Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.6.3 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:04.450550087Z" level=debug msg="configured \"token\" access controller" go.version=go1.6.3 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:04.450646341Z" level=info msg="listening on [::]:5000" go.version=go1.6.3 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
harbor-ui     | 2017-10-13T13:52:04Z [INFO] token expiration: 30 minutes
harbor-ui     | 2017-10-13T13:52:04Z [INFO] Config path: /etc/ui/app.conf
harbor-ui     | 2017-10-13T13:52:04Z [INFO] initializing database: type-MySQL host-mysql port-3306 user-root database-registry
harbor-ui     | 2017-10-13T13:52:04Z [ERROR] [mysql.go:79]: failed to connect to db, retry after 2 seconds :dial tcp 172.18.0.3:3306: getsockopt: connection refused
harbor-jobservice | 2017-10-13T13:52:05Z [INFO] Config path: /etc/jobservice/app.conf
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [config.go:95]: config: maxJobWorkers: 3
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [config.go:96]: config: localUIURL: http://ui
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [config.go:97]: config: localRegURL: http://registry:5000
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [config.go:98]: config: verifyRemoteCert: off
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [config.go:99]: config: logDir: /var/log/jobs
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [config.go:100]: config: uiSecret: ******
harbor-jobservice | 2017-10-13T13:52:05Z [INFO] token expiration: 30 minutes
harbor-jobservice | 2017-10-13T13:52:05Z [INFO] initializing database: type-MySQL host-mysql port-3306 user-root database-registry
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [workerpool.go:123]: worker 0 started
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [workerpool.go:123]: worker 1 started
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [workerpool.go:123]: worker 2 started
harbor-jobservice | 2017-10-13T13:52:05Z [DEBUG] [main.go:36]: Trying to resume halted jobs...
harbor-jobservice | 2017/11/13 13:52:05 [asm_amd64.s:1998][I] http server Running on :80
harbor-ui     | 2017-10-13T13:52:06Z [INFO] User id: 1 already has its encrypted password.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:245]: Start syncing repositories from registry to DB...
registry      | time="2017-10-13T13:52:06.71141096Z" level=debug msg="authorizing request" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=f34dc0a2-d243-4d81-a643-dbeaa47bb1aa http.request.method=GET http.request.remoteaddr="172.18.0.5:51618" http.request.uri="/v2/" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:06.712125865Z" level=warning msg="error authorizing context: authorization token required" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=f34dc0a2-d243-4d81-a643-dbeaa47bb1aa http.request.method=GET http.request.remoteaddr="172.18.0.5:51618" http.request.uri="/v2/" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | 172.18.0.5 - - [13/Oct/2017:13:52:06 +0000] "GET /v2/ HTTP/1.1" 401 87 "" "Go-http-client/1.1"
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [authutils.go:49]: scopes: [registry:catalog:*]
harbor-ui     | 2017-10-13T13:52:06Z [INFO] current access, type: registry, name:catalog, actions:[*]
registry      | time="2017-10-13T13:52:06.744199889Z" level=debug msg="authorizing request" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=a12e0319-2992-44e6-be19-7bec8e9ff926 http.request.method=GET http.request.remoteaddr="172.18.0.5:51622" http.request.uri="/v2/_catalog?n=1000" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:06.745186716Z" level=debug msg="filesystem.List(\"/docker/registry/v2/repositories\")" auth.user.name=admin go.version=go1.6.3 http.request.host="registry:5000" http.request.id=a12e0319-2992-44e6-be19-7bec8e9ff926 http.request.method=GET http.request.remoteaddr="172.18.0.5:51622" http.request.uri="/v2/_catalog?n=1000" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry trace.duration=51.277µs trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).List" trace.id=3731460c-4e17-430f-a985-465d28b298d2 trace.line=150 version=v2.5.0
registry      | time="2017-10-13T13:52:06.745614569Z" level=info msg="response completed" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=a12e0319-2992-44e6-be19-7bec8e9ff926 http.request.method=GET http.request.remoteaddr="172.18.0.5:51622" http.request.uri="/v2/_catalog?n=1000" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=2.61431ms http.response.status=200 http.response.written=20 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | 172.18.0.5 - - [13/Oct/2017:13:52:06 +0000] "GET /v2/_catalog?n=1000 HTTP/1.1" 200 20 "" "Go-http-client/1.1"
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [transport.go:51]: 200 | GET http://registry:5000/v2/_catalog?n=1000
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:297]: Start deleting repositories from DB...
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: devops/mqtt-cluster success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enter_one/dm_identity_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_datahub_mongo success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_dataviewer success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_dvservice success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_identity_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_userimage_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_web2mongo_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/experiment_boot_gen_170320 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/experiment_boot_gen_170320_3 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/message_player_service_choropleth success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/message_player_service_positions success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/vertx_kafkawebsocket_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/vertx_mqttmessaging_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: jhtest/dm_dvservice success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/haproxy success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/haproxy-mqtt-cluster success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/hello-world success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/mqtt-broker success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/mqtt-cluster-node-1 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/mqtt-cluster-node-2 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/nginx-int-geo-lb success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/pdf-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: lloesche/vertx_kafkawebsocket_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: lloesche/vertx_mqttmessaging_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/app-build-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/app-config-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/poi-editor-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/project-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/static-page-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: sbaust/node-microservice-sample success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: sbaust/node_microservice_sample success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: sbaust/spring-security-oauth-resource success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:307]: Sync repositories from registry to DB is done.
harbor-ui     | 2017/11/13 13:52:06 [asm_amd64.s:1998][I] http server Running on :80

This resulted in harbor being empty (o.O) and just showing us the projects without any data inside them.

For me, it looks like an error from the ui startup and the checking against the db and/or registry (var reposToDel in src/ui/api/utils.go maybe?!?):

.
.
harbor-ui     | 2017-10-13T13:52:04Z [INFO] initializing database: type-MySQL host-mysql port-3306 user-root database-registry
harbor-ui     | 2017-10-13T13:52:04Z [ERROR] [mysql.go:79]: failed to connect to db, retry after 2 seconds :dial tcp 172.18.0.3:3306: getsockopt: connection refused
.
.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:245]: Start syncing repositories from registry to DB...
registry      | time="2017-10-13T13:52:06.71141096Z" level=debug msg="authorizing request" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=f34dc0a2-d243-4d81-a643-dbeaa47bb1aa http.request.method=GET http.request.remoteaddr="172.18.0.5:51618" http.request.uri="/v2/" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:06.712125865Z" level=warning msg="error authorizing context: authorization token required" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=f34dc0a2-d243-4d81-a643-dbeaa47bb1aa http.request.method=GET http.request.remoteaddr="172.18.0.5:51618" http.request.uri="/v2/" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | 172.18.0.5 - - [13/Oct/2017:13:52:06 +0000] "GET /v2/ HTTP/1.1" 401 87 "" "Go-http-client/1.1"
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [authutils.go:49]: scopes: [registry:catalog:*]
harbor-ui     | 2017-10-13T13:52:06Z [INFO] current access, type: registry, name:catalog, actions:[*]
registry      | time="2017-10-13T13:52:06.744199889Z" level=debug msg="authorizing request" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=a12e0319-2992-44e6-be19-7bec8e9ff926 http.request.method=GET http.request.remoteaddr="172.18.0.5:51622" http.request.uri="/v2/_catalog?n=1000" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | time="2017-10-13T13:52:06.745186716Z" level=debug msg="filesystem.List(\"/docker/registry/v2/repositories\")" auth.user.name=admin go.version=go1.6.3 http.request.host="registry:5000" http.request.id=a12e0319-2992-44e6-be19-7bec8e9ff926 http.request.method=GET http.request.remoteaddr="172.18.0.5:51622" http.request.uri="/v2/_catalog?n=1000" http.request.useragent="Go-http-client/1.1" instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry trace.duration=51.277µs trace.file="/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go" trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).List" trace.id=3731460c-4e17-430f-a985-465d28b298d2 trace.line=150 version=v2.5.0
registry      | time="2017-10-13T13:52:06.745614569Z" level=info msg="response completed" go.version=go1.6.3 http.request.host="registry:5000" http.request.id=a12e0319-2992-44e6-be19-7bec8e9ff926 http.request.method=GET http.request.remoteaddr="172.18.0.5:51622" http.request.uri="/v2/_catalog?n=1000" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=2.61431ms http.response.status=200 http.response.written=20 instance.id=aef1c4ab-8c07-4a24-8a49-cea747beb340 service=registry version=v2.5.0
registry      | 172.18.0.5 - - [13/Oct/2017:13:52:06 +0000] "GET /v2/_catalog?n=1000 HTTP/1.1" 200 20 "" "Go-http-client/1.1"
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [transport.go:51]: 200 | GET http://registry:5000/v2/_catalog?n=1000
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:297]: Start deleting repositories from DB...
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: devops/mqtt-cluster success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enter_one/dm_identity_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_datahub_mongo success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_dataviewer success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_dvservice success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_identity_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_userimage_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/dm_web2mongo_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/experiment_boot_gen_170320 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/experiment_boot_gen_170320_3 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/message_player_service_choropleth success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/message_player_service_positions success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/vertx_kafkawebsocket_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: enterprise/vertx_mqttmessaging_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: jhtest/dm_dvservice success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/haproxy success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/haproxy-mqtt-cluster success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/hello-world success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/mqtt-broker success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/mqtt-cluster-node-1 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/mqtt-cluster-node-2 success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/nginx-int-geo-lb success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: library/pdf-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: lloesche/vertx_kafkawebsocket_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: lloesche/vertx_mqttmessaging_service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/app-build-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/app-config-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/poi-editor-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/project-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: microservices/static-page-service success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: sbaust/node-microservice-sample success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: sbaust/node_microservice_sample success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:302]: Delete repository: sbaust/spring-security-oauth-resource success.
harbor-ui     | 2017-10-13T13:52:06Z [DEBUG] [utils.go:307]: Sync repositories from registry to DB is done.

Did anyone ever face this problem before?
How can we get rid of it and make sure, that there will be no deletion when restarting harbor?

Best Regards,
Tim.

**Comment by reasonerjt** [comment]
Tim thanks for reporting.

When you see this error.
How many images do you have in your registry?

**Comment by klixxezitz** [comment]
Hej,

Running 'docker images'
shows the following result:

vmware/harbor-log          0.5.0               eebc987a891b        10 months ago       190.5 MB
vmware/harbor-jobservice   0.5.0               995368e96860        10 months ago       169.4 MB
vmware/harbor-ui           0.5.0               232a8664541a        10 months ago       233.1 MB
vmware/harbor-db           0.5.0               84c4ce8e9b6c        10 months ago       326.8 MB
nginx                      1.11.5              05a60462f8ba        11 months ago       181.5 MB
registry                   2                   7f4650c9d45e        14 months ago       33.28 MB
registry                   2.5.0               c6c14b3960bd        14 months ago       33.28 MB
nginx                      1.9                 c8c29d842c09        17 months ago       182.7 MB

Here is also the result of 'docker ps'
CONTAINER ID        IMAGE                            COMMAND                  CREATED             STATUS              PORTS                                      NAMES
35aaaeb9128a        nginx:1.11.5                     "nginx -g 'daemon off"   9 months ago        Up 2 days           0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   nginx
81ed4edbfdce        vmware/harbor-jobservice:0.5.0   "/harbor/harbor_jobse"   9 months ago        Up 2 days                                                      harbor-jobservice
5486fdbc7c91        vmware/harbor-ui:0.5.0           "/harbor/harbor_ui"      9 months ago        Up 2 days                                                      harbor-ui
ca7b0227bd0b        library/registry:2.5.0           "/entrypoint.sh serve"   9 months ago        Up 2 days           5000/tcp                                   registry
41b7e29bebdd        vmware/harbor-db:0.5.0           "docker-entrypoint.sh"   9 months ago        Up 2 days           3306/tcp                                   harbor-db
7c4be0cf4243        vmware/harbor-log:0.5.0          "/bin/sh -c 'crond &&"   9 months ago        Up 2 days           0.0.0.0:1514->514/tcp                      harbor-log



What I found out this weekend when checking the registry docker-compose.yml and the docker image via docker exec, is that our config seems to have an error.

When checking docker-compose.yml I see the following entry:
.
.
.
volumes:
      - /data/registryStorage:/storage
.
.
.
This means, if I got it right, that the local folder '/data/registryStorage' is mapped to '/storage' in the image, correct?
But when entering the image and checking '/storage' it is empty and all the registry data is in '/data/registryStorage' inside the image.

Might this cause the problem?
Because after a restart, '/data/registryStorage' inside the image is empty and the registry tells the startup process, that it has no data and to delete the repos from the database?!?

Best Regards,
Tim.
